srv/wordpress/wp-content/plugins/acore-wp-plugin/src/Hooks/WooCommerce/GuildChange.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace ACore\Hooks\WooCommerce;
+
+use ACore\Manager\ACoreServices;
+
+class GuildChange extends \ACore\Lib\WpClass {
+
+    public static function init() {
+        add_action('woocommerce_after_add_to_cart_quantity', self::sprefix() . 'before_add_to_cart_button');
+        add_filter('woocommerce_add_cart_item_data', self::sprefix() . 'add_cart_item_data', 20, 3);
+        add_filter('woocommerce_get_item_data', self::sprefix() . 'get_item_data', 20, 2);
+        add_action('woocommerce_checkout_order_processed', self::sprefix() . 'checkout_order_processed', 20, 2);
+        add_action('woocommerce_add_order_item_meta', self::sprefix() . 'add_order_item_meta', 1, 3);
+        add_action('woocommerce_payment_complete', self::sprefix() . 'payment_complete');
+    }
+
+    // 1) LIST
+    public static function before_add_to_cart_button() {
+        global $product;
+        if ($product->get_sku() != "guild-rename") {
+            return;
+        }
+
+        $current_user = wp_get_current_user();
+
+        if ($current_user) {
+            ?>
+            <p>You must select the guild master character.</p>
+            <?php
+            FieldElements::charList($current_user->user_login);
+            ?>
+            <br>
+            <br>
+            <label for="acore_new_guild_name">New name:</label>
+            <input type="text" maxlength="20" id="acore_new_guild_name" class="acore_new_guild_name" name="acore_new_guild_name">
```

Comment:
```suggestion
            <input type="text" maxlength="24" id="acore_new_guild_name" class="acore_new_guild_name" name="acore_new_guild_name">
```
To match values found in server code.